ISPN-11465 Upgrade to JGroups 4.2.1.Final

danberindei opened the pull request:
https://issues.redhat.com/browse/ISPN-11465

* Update the Infinispan and JGroups schema versions
  in the XML configurations
* Switch to FD_ALL3 in the test JGroups stacks
* Keep using FD_ALL in the default stacks

wburns commented on core/src/test/resources/configs/xsite/bridge.xml:
```diff
@@ -29,7 +29,11 @@
+    <FD_ALL3 timeout="3000"
+             interval="1000"
+    />
+    <VERIFY_SUSPECT timeout="1000"/>
```
So with FD_ALL3 it suspects the member first? I thought suspecting members was tossed out of JGroups a while back?

danberindei replied:
Not sure what you mean @wburns, it was always JGroups' responsibility to suspect members. X-site backups have their own rules for taking sites offline, but a site can have 0..N site masters, all of which join the bridge cluster. 

OTOH the bridge configuration already had {{FD_SOCK}}, which suspects a node once it's stopped, and tests don't remove {{FD*}} protocols from the bridge stack based on the {{TransportFlags}}, so having {{FD_ALL3}} might just complicate debugging.

True,  we have any xsite tests using {{DISCARD}} on the site masters, so {{FD_ALL3}} and {{VERIFY_SUSPECT}} aren't really needed, but they

danberindei replied:
Removed

wburns commented:
Weird I didn't see the xml parsing failures before https://ci.infinispan.org/job/Infinispan/job/PR-8052/3/#showFailuresLink

danberindei commented:
There were some conflicts with another PR that updated XSD versions and I didn't check that the merged stuff really works. I fixed the failures and updated more test configurations to use `FD_ALL3`.

wburns commented:
Integrated into master, thanks @danberindei !
